providers/base/bin/memory_compare.py

```diff
@@ -98,10 +174,15 @@ def main():
         return 1
 
     installed_memory = HumanReadableBytes(get_installed_memory_size())
+    fw_allocated_memory_for_pci = get_allocated_memory_for_each_pci_gpu();
     visible_memory = HumanReadableBytes(get_visible_memory_size())
     threshold = get_threshold(installed_memory)
 
-    difference = HumanReadableBytes(installed_memory - visible_memory)
+    if fw_allocated_memory_for_pci >= 0:
+        difference = HumanReadableBytes(installed_memory - visible_memory -
+                fw_allocated_memory_for_pci * 1024 * 1024)
```

Comment:
so you previously divide the size by 1024*1024 to once again multiply it here?
I recommend using basic units everywhere.